src/server.ts

```diff
@@ -1,48 +1,47 @@
 import express from 'express';
-import { readdirSync } from "fs";
-import { convertJatsToHtml } from "./conversion/encode";
 import { generateArticleList } from "./article-list/article-list";
 import { buildArticlePage } from "./article/article";
 import { generateReviewPage } from "./reviews/reviews";
 import { basePage } from "./base-page/base-page";
+import { StoreType, createArticleRepository, ArticleRepository } from './model/model';
+import { loadXmlArticlesFromDirIntoStores } from './data-loader/data-loader';
+import { createEnhancedArticleGetter, getEnhancedArticle } from './reviews/get-enhanced-article';
 
 
 const app = express();
-const cache: Record<string, string> = {};
 
-const getDirectories = (source: string) => {
-  return readdirSync(source, { withFileTypes: true })
-    .filter(dirent => dirent.isDirectory())
-    .map(dirent => dirent.name);
+const config = {
+  id: 'https://elifesciences.org',
+  name: "eLife",
+  dataDir: './data/10.1101',
+  databasePath: './data.db'
 }
 
+let articleRepository: ArticleRepository;
+let getEnhancedArticle: getEnhancedArticle;
+createArticleRepository(StoreType.Sqlite, config.databasePath).then((repo: ArticleRepository) => {
+  articleRepository = repo;
+  loadXmlArticlesFromDirIntoStores(config.dataDir, articleRepository);
+  getEnhancedArticle = createEnhancedArticleGetter(articleRepository, config.id);
+});
+
+
 app.use(express.static('public'));
 
-app.get('/', (req, res) => {
-  const articles: Record<string, Array<string>> = {};
-  const journals = getDirectories('./data');
-  journals.forEach(journal => articles[journal] = []);
-  journals.forEach(journal => {
-    getDirectories(`./data/${journal}`).forEach(articleDir => articles[journal].push(articleDir))
-  })
-  res.send(basePage(generateArticleList(journals, articles)));
+app.get('/', async (req, res) => {
```

Comment:
Since this and subsequent requests require `createArticleRepository` to resolve before they are executed how do we ensure that happens without using `await`.

Could we possibly perform the dependant requests in a `.then()`?